### ISSUE TITLE
Expose cryojax base classes

### ISSUE THREAD

**Issue by mjo22** [mjo22/cryojax#7]
Currently, `cryojax` abstract base classes are not exposed in the `cryojax.simulator.__init__.py` because their names are overridden by `Union` type hints.

This means that users cannot extend `cryojax` in their own workflows, such as writing new `Kernel`s, `Filter`s, or `Mask`s.

The best solution may be to instead use the abstract classes for type hints (this will be easier to maintain). This will break deserialization. To fix this, a `CryojaxObject` encoder can be added to `cryojax.core` that encodes the class name with the object dictionary, similar to the current ndarray encoder.
